src/DatePicker/index.tsx

```diff
@@ -1,68 +1,106 @@
+import { GERMAN_DATE_FORMAT } from '@mll-lab/js-utils';
+import generatePicker, {
+  PickerProps as AntdPickerProps,
+  PickerDateProps as AntdPickerDateProps,
+  PickerTimeProps as AntdPickerTimeProps,
+  RangePickerProps as AntdRangePickerProps,
+} from 'antd/es/date-picker/generatePicker';
+import 'antd/es/date-picker/style/index';
 import {
-  GERMAN_DATE_FORMAT,
-  GERMAN_DATE_TIME_FORMAT,
-  parseGermanDateFlexible,
-  MINIMAL_VALID_DATE,
-} from '@mll-lab/js-utils';
-import { addYears } from 'date-fns';
-import localeDe from 'date-fns/locale/de';
-import React from 'react';
-import {
-  ReactDatePickerProps,
-  registerLocale,
-  setDefaultLocale,
-} from 'react-datepicker';
+  format as formatDate,
+  parse as parseDate,
+  getWeek,
+  isValid,
+  startOfWeek,
+} from 'date-fns';
+import { de } from 'date-fns/locale';
+import dateFnsGenerateConfig from 'rc-picker/lib/generate/dateFns';
+import React, { ComponentClass } from 'react';
 
-import { StyledDatePicker } from './StyledDatePicker';
+type PickerProps<T> = AntdPickerProps<T>;
+type PickerDateProps<T> = AntdPickerDateProps<T>;
+type PickerTimeProps<T> = AntdPickerTimeProps<T>;
+type RangePickerProps<T> = AntdRangePickerProps<T>;
 
-registerLocale('de', localeDe);
-setDefaultLocale('de');
+export type DatePickerProps = PickerProps<Date> &
+  Omit<PickerDateProps<Date>, 'picker'>;
 
-export type DatePickerProps = Omit<ReactDatePickerProps, 'onChangeRaw'>;
+const localeParse = (format: string) =>
+  format
+    .replace(/Y/g, 'y')
+    .replace(/D/g, 'd')
+    .replace(/gggg/, 'yyyy')
+    .replace(/g/g, 'G')
+    .replace(/([Ww])o/g, 'wo');
 
-export function DatePicker(props: DatePickerProps) {
-  return (
-    <StyledDatePicker
-      {...props}
-      showYearDropdown
-      showMonthDropdown
-      scrollableYearDropdown
-      minDate={MINIMAL_VALID_DATE}
-      maxDate={addYears(new Date(), 1)}
-      yearDropdownItemNumber={50}
-      timeCaption="Zeit"
-      dateFormat={
-        props.showTimeSelect ? GERMAN_DATE_TIME_FORMAT : GERMAN_DATE_FORMAT
+export const BaseDatePicker: ComponentClass<PickerProps<Date>, any> & {
+  WeekPicker: ComponentClass<Omit<PickerDateProps<Date>, 'picker'>, any>;
```

Comment:
Will you fix these warnings, or is there a good reason for not fixing them right now?